app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/AddCoInsuredBottomSheetContent.kt

```diff
@@ -226,6 +233,37 @@ private fun FetchFromSsnFields(
   }
 }
 
+private fun personNummerInputFormatter(text: AnnotatedString, mask: String): TransformedText {
+  val trimmed = if (text.text.length >= 12) text.text.substring(0..11) else text.text
+
+  val annotatedString = AnnotatedString.Builder().run {
+    for (i in trimmed.indices) {
+      append(trimmed[i])
+      if (i == 7) {
+        append("-")
+      }
+    }
+    pushStyle(SpanStyle(color = Color.LightGray))
+    append(mask.takeLast(mask.length - length))
+    toAnnotatedString()
+  }
+
+  val personNummerOffsetTranslator = object : OffsetMapping {
+    override fun originalToTransformed(offset: Int): Int {
+      return when {
+        offset < 8 -> offset
+        offset <= 12 -> offset + 1
+        else -> 13
+      }
+    }
+
+    override fun transformedToOriginal(offset: Int): Int {
+      return text.length
+    }
```

Comment:
You may need to do the inverse logic here as well. I think otherwise the thing that looks odd is the cursor is on the wrong index, off by one. Can you test that to see how it is?

Reply:
If you try to highlight the text (by long pressing it, to copy it for example) and then you try to move the little handles around in order to select parts of the text, do those handles work as expected as well? That's what breaks usually I think if that mapping is not correct both back and forth.